manifests/init.pp

```diff
@@ -437,6 +432,16 @@
     }
   }
 
+  if $::osfamily == 'Suse' {
+    exec { 'install /etc/init.d/snmptrapd':
```

Comment:
Equivalent:
```puppet
file { '/etc/init.d/snmptrapd':
  source  => '/usr/share/doc/packages/net-snmp/rc.snmptrapd',
  owner   => 'root',
  group   => 'root',
  mode    => '0755',
  require => Package['snmpd'],
  before  => Service['snmptrapd'],
}
```
Possibly with a `replace => false` to emulate the `creates =>` on `exec`.

Reply:
PR updated.  I thought about the `replace => false` and decided it probably isn't needed.